actionpack/lib/abstract_controller/helpers.rb

```diff
@@ -150,21 +150,36 @@ def modules_for_helpers(args)
           case arg
           when String, Symbol
             file_name = "#{arg.to_s.underscore}_helper"
-            begin
-              require_dependency(file_name)
-            rescue LoadError => e
-              raise AbstractController::Helpers::MissingHelperError.new(e, file_name)
-            end
-
             mod_name = file_name.camelize
-            begin
-              mod_name.constantize
-            rescue LoadError
-              # dependencies.rb gives a similar error message but its wording is
-              # not as clear because it mentions autoloading. To the user all it
-              # matters is that a helper module couldn't be loaded, autoloading
-              # is an internal mechanism that should not leak.
-              raise NameError, "Couldn't find #{mod_name}, expected it to be defined in helpers/#{file_name}.rb"
+
+            if Rails.autoloaders.zeitwerk_enabled?
```

Comment:
This is coupling actionpack with railties. I think it would be better to inject this value in the railtie and use it here.